CIP-1694/README.md

```diff
@@ -0,0 +1,1161 @@
+
+#### Changes to the transaction body
+
+- [ ] New elements will be added to the transaction body, and existing update and MIR capabilities will be removed. In particular,
+
+  The governance actions and votes will comprise two new transaction body fields.
+
+- [ ] Three new kinds of certificates will be added in addition to the existing ones:
+
+  * DRep registration
+  * DRep de-registration
+  * Vote delegation
+
+  And similarly, the current MIR and genesis certificates will be removed.
+
+- [ ] A new `Voting` purpose will be added to Plutus script contexts.
+  This will provide, in particular, the vote to on-chain scripts.
+
+> **Warning** As usual, we will provide a CDDL specification for each of those changes.
+
+#### Changes to the existing ledger rules
+
+* The `PPUP` transition rule will be rewritten and moved out of the `UTxO` rule and into the `LEDGER` rule as a new `TALLY` rule.
+
+  It will process and record the governance actions and votes.
+
+* The `NEWEPOCH` transition rule will be modified.
+* The `MIR` sub-rule will be removed.
+* A new `RATIFY` rule will be introduced to stage governance actions for enactment.
+
+  It will ratify governance actions, and stage them for enactment in the current or next epoch, as appropriate.
+
+* A new `ENACTMENT` rule will be called immediately after the `EPOCH` rule. This rule will enact governance actions that have previously been ratified.
+* The `EPOCH` rule will no longer call the `NEWPP` sub-rule or compute whether the quorum is met on the PPUP state.
+
+#### Changes to the local state-query protocol
+
+The on-chain governance workload is large, but the off-chain workload for tools and applications will arguably be even larger.
+To build an effective governance ecosystem, the ledger will have to provide interfaces to various governance elements.
+
+While votes and DReps (de)registrations are directly visible in blocks and will, therefore, be accessible via the existing local-chain-sync protocols; we will need to upgrade the local-state-query protocol to provide extra insights on information which are harder to infer from blocks (i.e. those that require maintaining a ledger state). New state queries should cover (at least):
+
+- Governance actions currently staged for enactment
+- Governance actions under ratification, with the total and percentage of yes stake, no stake and abstain stake
+- The current constitutional committee, and constitution hash digest
+
+#### Bootstrapping Phase
+
+We will need to be careful how we bootstrap this fledgling government.  All the parties
+that are involved will need ample time to register themselves and to become familiar with the process.
+
+Special provisions will apply in the initial bootstrap phase.
+Firstly, during the bootstrap phase, a vote from the constitutional committee
+is sufficient to change the protocol parameters.
```

Comment:
Those special provisions would give the committee powers that were not present in the original draft written by the community before the workshop. 

While the original draft was built upon checks and balances, this change gives the committee, which has the weakest democratic representation of all three "chambers" (Stake/dReps, SPOs, Committee) the most power, even to unilaterally change protocol parameters.

I feel these major changes in the role of the committee have been introduced in a hush hush way due to the closed nature of the workshop, not been communicated enough and should be opened for discussion.

Reply:
In the same way that we did not instantly give SPOs the ability to produce all blocks at the moment of the Shelley hard fork (with the `d` mechanism), we want a smooth transition. This is a transitional phase to help facilitate a smooth transition. Nothing is "hush hush" about it, it is in this very public document.

Reply:
> I don't see why this helps a smooth transition.
> It can be quite the contrary. The committee is not specified well enough, e.g. in number of members.
> It would be even allowed to consist of a single person with no stake, for example a popular Youtuber.

The committee needs to be the group of experts today who understand how the protocol parameters effect the network. The hope is that they never have to make any governance actions at all, except to replace themselves. If there is an emergency, and we need to hard fork or change a protocol parameter, we do not want for enough DReps and to register and before we can act. Note that the interim committee has strictly fewer powers than the existing federated governance.

> This is not a smooth transition but an unnecessarily risky transition phase.

I disagree completely. This is a restriction of what we have today, with just enough powers for an emergency while we form a new system.

Reply:
> @JaredCorduan "The committee needs to be the group of experts today who understand how the protocol parameters effect the network."

1. Are you implying that the next CIP1694 specification revision will enforce a mechanism for genesis keyholder appointed constitutional committee(s).
2. Are you implying that constitutional committee member qualifications will be added to the next CIP1694 specification revision?

Reply:
> * Are you implying that the next CIP1694 specification revision will enforce a mechanism for genesis keyholder appointed constitutional committee(s).

We are still working out the details, but basically yes. It is my understanding that the interim committee will have new keys, but will probably selected from the known experts. It will likely be that the interim committee will have an expiration after a few months (entering no confidence if they cannot agree). We are also probably going to have per-member expiration so that we can have a more graceful rotational schedule.

> * Are you implying that constitutional committee member qualifications will be added to the next CIP1694 specification revision?

No, our legal experts are going to work on that in a separate document, as far as I understand.

Reply:
To avoid willful deception you must use a different name for an appointed committee such as [transitional, appointed, bootstrapping, central, genesis]-committee. IMHO "Constitutional" should be democratically reserved. The constitution hash committed to the chain would not be the result of a democratic process. This specification would require the SPO's and Drep's to hold a public vote of no confidence to have the opportunity to democratically elect a community-nominated constitutional committee to commit a democratic constitutional hash on-chain.

Emotionally, I don't think I can volunteer for Cardano anymore.
IOG isn't trusting the SPO's to serve Cardano governance during the bootstrapping phase.
I'm here for decentralization, democracy, the community, and separation of states & monies.
I didn't sign up for this. I'm going to go hang out with Liberlion.
Please use a tau-prime parameter to hard-prevent treasury drain.
If not, I'll be watching for treasury drain governance actions to liquidate my bag first.


I'll likely be back if:
1. A CIP1694 constitutional committee is elected by the Drep's and SPO's.
2. The community creates a constitution and commits the hash on chain.
3. The minPoolCost parameter is voted to 0.
4. The k parameter is changed.
5. The a0 parameter is lowered.
6. IOG operates only 1 privately pledged pool, delegates remaining stake, and retires all other pools.
7. Chain-bootstrap-from-genesis is live on mainnet so bootstrap-from-IOG-servers aren't necessary.

DM's always open.

Reply:
I think we were going to call it the "interim committee". What would count as the source of truth for the name of something, for you? This CIP?

> IOG isn't trusting the SPO's to serve Cardano governance during the bootstrapping phase.
> I'm here for decentralization, democracy, the community, and separation of states & monies.

The interim committee is supposed to be the transition from what we have today to complete handover. I'd say there is a lot of trust. They will have a short expiration, and after that the only protocol level option is for the DReps and the SPOs to vote yes on who they want (or say stuck in no confidence).